sonata/main.py

```diff
@@ -1234,8 +1234,12 @@ def on_add_item(self, _widget, play_after=False):
             if self.current_tab == self.TAB_LIBRARY:
                 items = self.library.get_path_child_filenames(True)
                 self.mpd.command_list_ok_begin()
-                for item in items:
-                    self.mpd.add(item)
+                for item in items:     ######
+                     if item.endswith('.pls') or item.endswith('.PLS') or \
```

Comment:
On 21.06.2015 20:42, Jonathan Ballet wrote:
>
> In sonata/main.py 
> <https://github.com/multani/sonata/pull/85#discussion_r32896050>:
>
> > @@ -1234,8 +1234,12 @@ def on_add_item(self, _widget, play_after=False):
> >              if self.current_tab == self.TAB_LIBRARY:
> >                  items = self.library.get_path_child_filenames(True)
> >                  self.mpd.command_list_ok_begin()
> > -                for item in items:
> > -                    self.mpd.add(item)
> > +                for item in items:     ######
> > +                     if item.endswith('.pls') or item.endswith('.PLS') or \
>
> This should be extracted into a dedicated function, since I guess 
> we'll have to support something else than |.pls| and |.m3u| playlists 
> one day.
> Also, this could be simplified if you convert your "item" value into 
> lower/upper case.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/multani/sonata/pull/85/files#r32896050>.
>
I can do that but like you said there must be a better way to discern 
between mp3's and pls's.

I can not remember that I somewhere found code that looks at the 
extension and says: this is not an mp3 and therefore should not be shown 
in the fileview window. Do you?